### ISSUE TITLE
failed to catalog: could not fetch image (only on v0.47.0)

### ISSUE THREAD

**Issue by DanFainshteinGC** [anchore/grype#882]
**What happened**:
When trying to scan an image on v0.47.0, I get the following error:
```
1 error occurred:
	* failed to catalog: could not fetch image "5428983d1a2b": unable to use DockerDaemon source: pull failed: Error response from daemon: pull access denied for 5428983d1a2b, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```
The image is a local image. It works perfectly on v0.46.0.

**What you expected to happen**:
The image should've been scanned.

**How to reproduce it (as minimally and precisely as possible)**:
Just run "grype container_id" on a locally stored image.

**Anything else we need to know?**:
Another piece of information that might help, is that I got the same image on Docker on a remote registry.
I.E Docker output:
```
172.16.0.1:5000/redis                 v44_1         5428983d1a2b   4 months ago    38.1MB
redis                                 v44_1         5428983d1a2b   4 months ago    38.1MB
```
And I'm trying to scan the second one.

**Environment**:
- Output of `grype version`: 
`Application:          grype
Version:              0.47.0
Syft Version:         v0.54.0
BuildDate:            2022-08-17T20:00:45Z
GitCommit:            08b4ef493b36a65f6149c9092d083d5d57540cdc
GitDescription:       v0.47.0
Platform:             linux/amd64
GoVersion:            go1.18.5
Compiler:             gc
Supported DB Schema:  4`
- OS (e.g: `cat /etc/os-release` or similar): Ubuntu 18.04 LTS

**Comment by kzantow** [comment]
@DanFainshteinGC are you running grype like this, with just the image hash?
```
grype 5428983d1a2b
```

**Comment by kzantow** [comment]
A few notes, I did some digging about what changed, basically a Syft upgrade:
https://github.com/anchore/grype/compare/v0.46.0...anchore:v0.47.0
Which included these changes:
https://github.com/anchore/syft/compare/v0.53.4...anchore:v0.54.0
And these stereoscope changes:
https://github.com/anchore/stereoscope/compare/c55b13fee7e4...anchore:84004345484e
I wonder if the `latest` default might have something to do with this, does it work for you if you include the tag @DanFainshteinGC? E.g.:
```
grype 5428983d1a2b:v44_1
```

**Comment by DanFainshteinGC** [comment]
@kzantow Yes, I run it with the image hash for a reason:
I have a bash script that scans all the images on a machine (as a workaround to the lack of "--all" flag, maybe I should try to implement it 😄). Originally, the issue I mentioned here happened to me when trying to scan with "grype redis" because it defaulted to the remote image I can't acquire, so I started scanning with the image hash which made Grype scan the local image. 
------
On another note, yes! It works when mentioning the version tag! Is version tag requirement by design? if so, maybe we can add a more informative error message. Otherwise, it's a regression from v0.46.0..

**Comment by freedom-isnotanarchy** [comment]
You write, ((   (as a workaround to the lack of "--all" flag, maybe I should try to implement it 😄). ))
I too, enjoy the **"--all" flag functionality**.
_______
grype **--help**
A vulnerability scanner for container images, filesystems, and SBOMs.

Supports the following image sources:
    grype yourrepo/yourimage:tag             defaults to using images from a Docker daemon
    grype **path/to/yourproject**                a Docker tar, OCI tar, OCI directory, **or generic filesystem directory**
    grype attestation.json --key cosign.pub  extract and scan SBOM from attestation file

**Comment by kzantow** [comment]
@DanFainshteinGC -- I double-checked with the team and the behavior here has changed intentionally. The problem you have is the _short_ hash which is not clear that it is, in fact, an image hash or just a randomized image name. If you use the full image hash, something like `grype sha256:bf943ccda095bf03693103e5249c1dac1ee1787101246a7e94338958ba2f1f5f`, then this should work fine -- are you able to modify scripts to get full image hashes to use?

**Comment by kzantow** [comment]
@freedom-isnotanarchy -- PRs happily accepted! 😄

**Comment by DanFainshteinGC** [comment]
@kzantow Yes, it works with full image hash and I am able to modify my script to display them (just add --no-trunc to "docker image ls" command).

**Comment by kzantow** [comment]
@DanFainshteinGC -- great news! In that case I'm going to close this issue. Of course, please reopen if there is something I'm misunderstanding!

I've split the request for scanning all local images to: #909, if this is something still desired please leave comments there so we can try to come up with a plan!
